.github/workflows/release_parallel.yml

```diff
@@ -164,46 +140,74 @@ jobs:
               with:
                   ref: ${{ github.event.pull_request.head.ref || github.ref }}
                   token: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
-                  fetch-depth: 0 # Fetch all history for all branches and tags
+                  fetch-depth: 0
 
-            - name: Merge update branches
+            - uses: ./.github/actions/configure_git
+              with:
+                  gpg_signing_key: ${{ secrets.GPG_SIGNING_KEY }}
+
+            - name: Collect branches
+              id: collect-branches
+              run: |
+                  # Create array of branches
+                  readarray -t BRANCH_ARRAY < <(git branch -r | grep "origin/update-${{ needs.prepare-release.outputs.release-version }}-" | sed 's/origin\///' | xargs -n1)
+
+                  # Output array elements as multiline value
+                  echo "branches<<EOF" >> $GITHUB_OUTPUT
+                  printf '%s\n' "${BRANCH_ARRAY[@]}" >> $GITHUB_OUTPUT
+                  echo "EOF" >> $GITHUB_OUTPUT
+
+            - name: Display collected branches
+              run: |
+                  echo "Collected branches:"
+                  while read branch; do
+                    echo "  - $branch"
+                  done < <(echo "${{ steps.collect-branches.outputs.branches }}")
+                  echo "End of branch list"
+
+            - name: Fetch branches
+              run: |
+                  echo "Fetching all branches..."
+                  BRANCHES_TO_FETCH=""
+                  while read branch; do
```

Comment:
Use for loops and single space separated lists

Reply:
done